src/reporter.js

```diff
@@ -8,14 +9,30 @@ module.exports.reportToS3 = async function (
     functionName,
     awsRequestId
 ) {
-    if (!s3Client) {
-        const S3 = require('aws-sdk/clients/s3')
-        s3Client = new S3()
-    }
     const params = {
         Body: JSON.stringify(profilingData),
         Bucket: bucketName,
         Key: `${pathPrefix}${functionName}/${awsRequestId}/${fileName}.cpuprofile`,
     }
-    return s3Client.putObject(params).promise()
+
+    if (!s3Client) {
+        if (isSDKV3()) {
+            const { S3Client, PutObjectCommand } = require('@aws-sdk/client-s3'); 
+            s3Client = new S3Client();
+            putObjectCommand = new PutObjectCommand(params);
```

Comment:
We should not cache PutObjectCommand object and create new one at each report, because S3 params (key because of request id) are changed at each report.

So I think, you should remove the `PutObjectCommand` object creation here and create it while sending command as following:
```node
return isSDKV3()
        ? s3Client.send(new PutObjectCommand(params))
        : s3Client.putObject(params).promise();
```